### ISSUE TITLE
`group` attribute spoils `markersize`

### ISSUE THREAD

**Issue by heliosdrm** [JuliaPlots/Plots.jl#962]
If data points of a scatter plot are grouped by the `group` attribute, mapping the marker size to a variable does not work properly -- except in the first group. 

See small example:

```julia
v = 1:10
g = repeat(["A","B"], inner=5)
scatter(v,v, markersize=2v, color=:lightgrey)
scatter!(v,v, markersize=2v, group=g)
```

Output:
![newplot](https://user-images.githubusercontent.com/10651028/27806911-b8b961fa-603d-11e7-959f-6b11b388c9ef.png)

Grey circles have the correct size; the reddish points ("A") just overlap the grey ones, but the green ones ("B") have a different scale.

**Comment by heliosdrm** [comment]
However, when the data series are manually split, instead of using the `group` attribute, the behaviour is the one I expected (at least in this case). Namely

```julia
scatter(v[g.=="A"],v[g.=="A"], markersize=2v[g.=="A"])
scatter!(v[g.=="B"],v[g.=="B"], markersize=2v[g.=="B"])
```

![newplot](https://user-images.githubusercontent.com/10651028/28693868-29874112-7327-11e7-9438-9cff7470a34b.png)

My intuition is that both ways of grouping the data should provide the same result. In this particular case I'm grateful that it doesn't, so that I can choose how to define the size of the markers. But this looks like hacking an ambiguous behaviour, rather than the proper way of doing it.

I suspect that the size of the markers in the legend has something to do with this issue, but I don't understand how this works.

**Comment by heliosdrm** [comment]
Ok, forget the previous comment. I've just understood what you meant.

Anyway, for what is worth, I would expect that `group` splits all vector arguments that are as long as the data series `x`, `y`, etc.

**Comment by mkborregaard** [comment]
I see what you mean, that could work, I think (but wouldn't be trivial to implement).

**Comment by piever** [comment]
Closed by #1049
